api/core/workflow/nodes/trigger_webhook/node.py

```diff
@@ -112,13 +120,42 @@ def _get_normalized(mapping: dict[str, Any], key: str) -> Any:
 
             if param_type == "file":
                 # Get File object (already processed by webhook controller)
-                file_obj = webhook_data.get("files", {}).get(param_name)
-                outputs[param_name] = file_obj
+                files = webhook_data.get("files", {})
+                if files and isinstance(files, dict):
+                    file = files["file"]
+                    if file and isinstance(file, dict):
+                        try:
+                            related_id = file.get("related_id")
+                            transfer_method_value = file.get("transfer_method")
+                            if not transfer_method_value:
+                                raise ValueError("transfer_method is required in file mapping")
+                            transfer_method = FileTransferMethod.value_of(transfer_method_value)
+                            match transfer_method:
+                                case FileTransferMethod.LOCAL_FILE | FileTransferMethod.REMOTE_URL:
+                                    file["upload_file_id"] = related_id
+                                case FileTransferMethod.TOOL_FILE:
+                                    file["tool_file_id"] = related_id
+                                case FileTransferMethod.DATASOURCE_FILE:
+                                    file["datasource_file_id"] = related_id
+                            file_obj = file_factory.build_from_mapping(
+                                mapping=file,
+                                tenant_id=self.tenant_id,
+                            )
+                            file_segment = build_segment_with_type(SegmentType.FILE, file_obj)
+                            outputs[param_name] = FileVariable(
+                                name=param_name, value=file_segment.value, selector=[self.id, param_name]
+                            )
+                        except ValueError:
```

Comment:
Does this except branch correspond to the above `raise ValueError("transfer_method is required in file mapping")`. If it is, I would suggest change here too.

Since `ValueError` is also used by Python runtime and third-party libraries, catching it could hide exceptions caused by user code and incorrect usage of third-party libraries.